experimental/src/Config/ConfigBuilder.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Experimental\Config;
+
+use OpenTelemetry\Experimental\Config\Exporter\Otlp\Config as OtlpConfig;
+use OpenTelemetry\Experimental\Config\SpanProcessor\BatchSpanProcessorConfig;
+use OpenTelemetry\Experimental\Config\SpanProcessor\NoopSpanProcessorConfig;
+use OpenTelemetry\Experimental\Config\SpanProcessor\SimpleSpanProcessorConfig;
+
+class ConfigBuilder
+{
+    //initial list of known exporter types (exluding exotic contrib exporters, which must be added)
+    private array $exporters = [
+        OtlpConfig::class,
+    ];
+    //initial list of known span processor types
+    private array $spanProcessors = [
+        BatchSpanProcessorConfig::class,
+        SimpleSpanProcessorConfig::class,
+        NoopSpanProcessorConfig::class,
+    ];
+    private array $userConfig = [];
+    private array $environmentConfig = [];
+
+    public function __construct()
+    {
+        $this->environmentConfig = $this->getEnvironmentConfig();
+    }
+
+    public function build(): object
+    {
+        $config = new Config($this->userConfig, $this->environmentConfig);
+        foreach ($this->buildExporters() as $name => $exporterConfig) {
```

Comment:
What#s the point of passing the whole config tree into dedicated configs? What's the point of having them then in the first place, as this is hard coupling them with the global config schema.

Reply:
I think I see an improvement here. I think that the config builder should creates all of the different sub-configs and assemble them, passing to each only the config that they need.